src/bucket/BucketListBase.cpp

```diff
@@ -127,13 +129,82 @@ template <typename BucketT>
 void
 BucketLevel<BucketT>::commit()
 {
-    if (mNextCurr.isLive())
+    if (mNextCurrInMemory)
     {
+        // If we have an in-memory merged result, use that
+        setCurr(mNextCurrInMemory);
+    }
+    else if (mNextCurr.isLive())
+    {
+        // Otherwise use the async merge result
         setCurr(mNextCurr.resolve());
     }
     releaseAssert(!mNextCurr.isMerging());
 }
 
+template <>
+template <typename... VectorT>
+void
+BucketLevel<LiveBucket>::prepareFirstLevel(Application& app,
+                                           uint32_t currLedger,
+                                           uint32_t currLedgerProtocol,
+                                           bool countMergeEvents, bool doFsync,
+                                           VectorT const&... inputVectors)
+{
+    ZoneScoped;
+
+    // This method should only be called on level 0
+    releaseAssert(mLevel == 0);
+
+    // We shouldn't have an in-progress merge
+    releaseAssert(!mNextCurr.isMerging());
+    releaseAssert(!mNextCurrInMemory);
+
+    auto curr =
+        BucketListBase<LiveBucket>::shouldMergeWithEmptyCurr(currLedger, mLevel)
+            ? std::make_shared<LiveBucket>()
+            : mCurr;
+
+    // On startup, the current bucket may not be initialized in memory, so
+    // fallback to normal prepare
+    if (!curr->hasInMemoryEntries())
+    {
+        auto snap = LiveBucket::fresh(
+            app.getBucketManager(), currLedgerProtocol, inputVectors...,
+            countMergeEvents, app.getClock().getIOContext(), doFsync);
+        prepare(app, currLedger, currLedgerProtocol, snap, /*shadows=*/{},
+                countMergeEvents);
+        return;
+    }
+
+    auto snap = LiveBucket::fresh(
+        app.getBucketManager(), currLedgerProtocol, inputVectors...,
+        countMergeEvents, app.getClock().getIOContext(), doFsync,
+        /*storeInMemory=*/true, /*shouldIndex=*/false);
```

Comment:
`shouldIndex` is false here because indexing is redundant: we merge this newly created bucket right away and create a new (persistent) index anyway. Is that correct? I think a comment with some explanation would be nice here.